dotnet/src/Planners/Planners.Handlebars/Handlebars/HandlebarsPlan.cs

```diff
@@ -46,11 +48,28 @@ public override string ToString()
     /// <param name="arguments">The arguments.</param>
     /// <param name="cancellationToken">The cancellation token.</param>
     /// <returns>The plan result.</returns>
-    public string Invoke(
+    public async Task<string> InvokeAsync(
         Kernel kernel,
         KernelArguments arguments,
         CancellationToken cancellationToken = default)
     {
-        return HandlebarsTemplateEngineExtensions.Render(kernel, this._template, arguments, cancellationToken);
+        var logger = kernel.LoggerFactory.CreateLogger(typeof(HandlebarsPlan));
+
+        return await PlanInstrumentation.InvokePlanAsync(
+            static (HandlebarsPlan plan, Kernel kernel, KernelArguments arguments, CancellationToken cancellationToken)
+                => plan.InvokeCoreAsync(kernel, arguments, cancellationToken),
+            this, kernel, arguments, logger, cancellationToken).ConfigureAwait(false);
+    }
+
+    private async Task<string> InvokeCoreAsync(
+        Kernel kernel,
+        KernelArguments arguments,
+        CancellationToken cancellationToken = default)
+    {
+        Task<string> task = Task.Run(
+            () => HandlebarsTemplateEngineExtensions.Render(kernel, this._template, arguments, cancellationToken),
+            cancellationToken);
+
+        return await task.ConfigureAwait(false);
```

Comment:
This shouldn't use Task.Run.  If the core of this is synchronous, why make everything Task-returning?